docs/sphinx/source/contributing/style_guide.rst

```diff
@@ -70,6 +70,77 @@ the ``docs/readthedocs.org:pvlib-python`` link within the checks
 status box at the bottom of the pull request.
 
 
+.. _references:
+
+References
+----------
+pvlib-python is transitioning to the `IEEE reference style <https://journals.ieeeauthorcenter.ieee.org/wp-content/uploads/sites/7/IEEE_Reference_Guide.pdf>`_,
+which should be used in all new contributions. At minimum, a reference should
+include:
+
+* Author list (for more than six authors, list only the first with et al.)
+* Publication title
+* Publisher (journal title, laboratory name, etc.)
+* Year of publication
+* DOI (if available)
+
+The recommended citation style for several media types is as follows:
+
+**Journal article**:
+
+    Author initials and surname, "Title of article," abbreviated journal
+    title, vol. number, issue number, page numbers, Abbreviated Month Year.
+
+**Book**:
+
+    Author initials. Author Surname, "Chapter title" in Book Title, xth ed.
+    City of Publisher, (only U.S. State), Country: Abbrev. of Publisher, year,
+    ch. x, sec. x, pp. xxx–xxx.
+
+**Technical report**:
+
+    Author initials. Author Surname, "Report title" Abbrev. publisher name,
+    City of publisher, Abbrev. State, Country, Rep. xxx, year
+
+The example below shows how to cite material and generate a reference list
+using the IEEE style in a docstring::
+
+    This is the recommended citation for the pvlib-python project [1]_. There
+    are also some conference papers linked to pvlib, for example [2]_.
+    
+    Other types of reference you may find in the pvlib-python documentation
+    include books [3]_ and technical reports [4]_.
+
+    References
+    ----------
+    .. [1] K. Anderson, C. Hansen, W. Holmgren, A. Jensen, M. Mikofski,
+           and A Driesse. "pvlib python: 2023 project update." J. Open Source
+           Softw. vol. 8, no. 92, pp. 5994, Dec 2023,
+           :doi:`10.21105/joss.05994`
+
+    .. [2] J. S. Stein, "The Photovoltaic Performance Modeling Collaborative
+           (PVPMC)," In Proc. 38th IEEE Photovoltaic Specialists Conference
+           (PVSC), Austin, TX, USA, 2012, pp. 3048-3052,
+           :doi:`10.1109/PVSC.2012.6318225`
+
+    .. [3] J. A. Duffie and W. A. Beckman, "Solar Radiation" in Solar
+           Engineering of Thermal Processes, 3rd ed, New York, USA, J. Wiley
+           and Sons, 2006, ch. 1, sec. 1.5, pp.9-11
+
+    .. [4] R. Bird and C. Riordan, "Simple solar spectral model for direct and
+           diffuse irradiance on horizontal and tilted planes at the earth’s
+           surface for cloudless atmospheres", NREL, Golden, CO, USA, Technical
+           Report TR-215-2436, 1984, :doi:`10.2172/5986936`
+
+Things to note:
+
+* In-text numeric citations are a number inside square brackets, followed
+  by an underscore, e.g. ``[1]_``.
+* To include a DOI, you can use the existing ``:doi:``
+  `Sphinx role <https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html>`_,
+  followed by the DOI string inside a set of backticks.
```

Comment:
I wonder whether this link is confusing since `:doi:` role does not appear in that page.

```suggestion
* To include a DOI link, use the custom ``:doi:`` role, e.g.,
  |:doi:`10.21105/joss.05994`| will render as `DOI: 10.21105/joss.05994 <dx.doi.org/10.21105/joss.05994>`_.
```

Reply:
> I wonder whether this link is confusing since `:doi:` role does not appear in that page.

I think it's okay if `:doi:` is not there as only phrase "sphinx role" is linked; the purpose was just to provide some context/info in case the reader is unfamiliar with the term "role" --- I accept it is not essential, but I was just trying to be helpful 😅

I am not opposed to revising further, but I am unsure about the suggested revision... Save for the "render as..." I think this is covered by the example?

Reply:
At least add the word `custom` in front of it, please.

```suggestion
* To include a DOI, you can use pvlib's custom ``:doi:``
  `Sphinx role <https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html>`_,
  followed by the DOI inside a set of backticks.
```